### DISCUSSION TITLE
API layer in Nuxt by default

### DISCUSSION THREAD

**Discussion by splincool** [nuxt/nuxt#17087] [Ideas]
Nuxt has a lot of cool features by default (like pages, plugins, middleware etc.). What if It would be some kind of recommended way of fetching data from a server, like an API layer. For example in Angular they have services and the DI pattern. In Nuxt it could be provide/inject or maybe something else.

I have seen a lot of projects where teams create their own ways and methods to fetch data from a server. Some ways were good, some ways were bad and overcomplicated. But if it would be provided by default in my opinion it would help a lot.

When you switch between Nuxt projects (within one company or some companies), you have already know how to create pages in new Nuxt projects or plugins or middleware..., because it goes from the framework and from the docs.  And on top of that you will know how to create an API layer that is usually one of the important part of a project.

**Comment by Angloya** [comment]
It's cool idea
